keps/sig-cli/kustomize-secret-generator-plugins.md

```diff
@@ -205,52 +266,114 @@ are out of scope for this KEP, however this KEP's
 implementation will make it much easier to create other
 kinds of plugins.
 
-Go Plugins that become popular have a clear
-path to becoming a builting - so if someone writes, say,
-a general Exec plugin, we can easily promote it to a
-builtin (by virtue of the fact that it's written in Go,
-and because of the choices made in this KEP for
-describing a plugin in the kustomization file).
+### goplugin limitations
+
+The first plugin mechanism will be goplugins, since by
+design they are trivial to make for a Go program (like
+kustomize).  However, they have limitations, and are
+to some  extent an experimental feature of Go tooling.
+
+#### Not shareable as object code
+
+The shared object (`.so`) files created via `go build
+-buildmode=plugin` cannot be reliably used by a loading
+program unless both the program and the plugin were compiled
+with the same version of Go, with the same libs, on the same
+machine (see, e.g. [this golang issue](https://github.com/golang/go/issues/18827)).
+
+Therefore, the Go developer who elects to write their
+key:value generator as a goplugin is obligated to compile a
+new kustomize binary as well as the plugin.
+
+For this developer, this extra compilation step will be far
+easier than forking kustomize and modifying the code
+directly to introduce all the piping one would need to
+express a new key:value secret generation style in a
+kustomization file and integrate it with existing methods.
+
+The risk to not compiling, and, say using an `.so` that was
+compiled long ago and far away is that the program will
+panic when the `build` command is executed.
+
+##### mitigation
+
+To remind users of this risk, a special runtime
+flag will be required; see _graduation criteria_ below.
+
+##### No current support for Windows
 
-#### No current windows support for golang plugins.
-This may be implemented by the Go team later in which
-case it will just work.
+This is mentioned in the
+[plugin package overview section](https://golang.org/pkg/plugin).
 
-Also, as noted above, someone can write an exec style plugin
+##### mitigation
+
+This KEP offers a framework that dramatically
+lowers the barrier to someone writing an exec style plugin
 which will work on Windows.
 
+Of course, Windows users who want to support
+goplugins could contribute to the golang project.
+
 #### General symbol execution from the plugin
-A Go plugin will be cast to a particular hardcoded interface, e.g.
+
+##### mitigation
```

Comment:
We could add that one alternative considered is to disable this feature and require an ldflag to be set at build time to enable it, but we aren't doing this because it would reduce transparency when the command was run about whether the feature was enabled or not.